core/src/apps/ripple/serialize.py

```diff
@@ -81,9 +82,9 @@ def write_type(w: bytearray, field: dict):
 
 def serialize_amount(value: int) -> bytearray:
     if value < 0:
-        raise ValueError("Only non-negative integers are supported")
+        raise wire.DataError("Only non-negative integers are supported")
```

Comment:
This seems like a particularly wrong place for a `wire.Error`. This is a deep utility function in a module that doesn't know anything about the wire. IMHO `ValueError` is very appropriate here, and if we want to be nice about the problems, the values should have been checked in caller code

Reply:
I know. And on one side I agree, but on the other the same argument could be used for `helpers`, `common` functions etc. And having each such file raising `ValueError` and then catching seems like an actual overhead while achieving very little, doesn't it?

Reply:
that's only in case you want to call the function and re-raise the ValueError.

i'd argue that a better style here and in similar cases is to have the `ValueError` indicate an _exceptional_ state, and make sure the calling code had already validated the data for user-reportable problems

Reply:
I see. So you are basically saying I should leave `ValueError` here and check if `value < 0` is in `sing_tx.validate` beforehand. I like that a lot!